pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	return gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+}
+
+func (gqm *GroupQuotaManager) getGroupQuotaInfoByNameNoLock(quotaName string) *QuotaInfo {
+	return gqm.quotaInfoMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) getTreeWrapperByNameNoLock(quotaName string) (treeWrapperPtr *QuotaTreeWrapper) {
+	return gqm.quotaTreeWrapperMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) scopedLockForQuotaInfo(quotaList []*QuotaInfo) func() {
+	listLen := len(quotaList)
+	for i := listLen - 1; i >= 0; i-- {
+		quotaList[i].lock.Lock()
+	}
+
+	return func() {
+		for i := 0; i < listLen; i++ {
+			quotaList[i].lock.Unlock()
+		}
+	}
+}
+
+func (gqm *GroupQuotaManager) DealWithQuota(quota *v1alpha1.ElasticQuota, isDelete bool) error {
+	gqm.Lock.Lock()
+	defer gqm.Lock.Unlock()
+
+	quotaName := quota.Name
+	inputQuotaGroups := make([]*QuotaInfo, 0)
+	if isDelete {
+		_, exist := gqm.quotaInfoMap[quotaName]
+		if !exist {
+			return fmt.Errorf("get quota info failed, quotaName:%v", quotaName)
+		}
+		delete(gqm.quotaInfoMap, quotaName)
+	} else {
+		quotaInfo := NewQuotaInfoFromQuota(quota)
+		// if update/create DefaultQuotaGroup, no need to resetTree.
+		if quotaName == extension.DefaultQuotaName {
+			quotaInfo.CalculateInfo.Used = gqm.quotaInfoMap[extension.DefaultQuotaName].GetUsed()
+			quotaInfo.CalculateInfo.Request = gqm.quotaInfoMap[extension.DefaultQuotaName].GetRequest()
+			quotaInfo.CalculateInfo.Runtime = gqm.quotaInfoMap[extension.DefaultQuotaName].GetRuntime()
+			gqm.quotaInfoMap[extension.DefaultQuotaName] = quotaInfo
+			return nil
+		}
+
+		inputQuotaGroups = append(inputQuotaGroups, quotaInfo)
+	}
+	return gqm.updateQuotaGroupConfigNoLock(inputQuotaGroups)
+}
+
+func (gqm *GroupQuotaManager) updateQuotaGroupConfigNoLock(inputGroups []*QuotaInfo) error {
+	//build allNodes based on the input
+	newAllQuotaTopoNodes := gqm.buildSubParGroupTopoNoLock(inputGroups)
+
+	gqm.resetAllGroupQuotaNoLock(newAllQuotaTopoNodes)
+	return nil
+}
+
+//BuildSubParGroupTopoNoLock reBuild a nodeTree from root, no need to lock gqm.lock
+func (gqm *GroupQuotaManager) buildSubParGroupTopoNoLock(updateGroups []*QuotaInfo) map[string]*QuotaTopoNode {
+	//rebuild tree
+	allQuotaTopoNodes := make(map[string]*QuotaTopoNode)
+	rootNode := NewQuotaTopoNode(NewQuotaInfo(true, true, extension.RootQuotaName, extension.RootQuotaName))
+	allQuotaTopoNodes[extension.RootQuotaName] = rootNode
+
+	//get already created node
+	for quotaName, quotaInfo := range gqm.quotaInfoMap {
+		allQuotaTopoNodes[quotaName] = NewQuotaTopoNode(quotaInfo)
+	}
+	//get newCreated nodes
+	for _, quotaInfo := range updateGroups {
+		if _, exist := allQuotaTopoNodes[quotaInfo.Name]; !exist {
+			topoNode := NewQuotaTopoNode(&QuotaInfo{
+				Name: quotaInfo.Name,
+			})
+			allQuotaTopoNodes[quotaInfo.Name] = topoNode
+		}
+	}
+	for _, quotaInfo := range updateGroups {
```

Comment:
the second for can be killed based on the above annotation: UpdateQuotaInfoFromRemote has been executed before.